plaso/engine/artifact_filters.py

```diff
@@ -75,71 +74,111 @@ def BuildFindSpecs(self, environment_variables=None):
     """
+    find_specs = []
     for name in self._artifacts:
       definition = self._artifacts_registry.GetDefinitionByName(name)
       if not definition:
         continue
 
-      for source in definition.sources:
-        if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
-          # TODO: move source.paths iteration into
-          # BuildFindSpecsFromFileArtifact.
-          for path_entry in set(source.paths):
-            find_specs = self.BuildFindSpecsFromFileArtifact(
-                path_entry, source.separator, environment_variables,
-                self._knowledge_base.user_accounts)
-            artifact_group = self._find_specs_per_source_type[
-                artifact_types.TYPE_INDICATOR_FILE]
-            artifact_group.extend(find_specs)
-
-        elif (source.type_indicator ==
-              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
-          # TODO: move source.keys iteration into
-          # BuildFindSpecsFromRegistryArtifact.
-          for key_path in set(source.keys):
-            if self.CheckKeyCompatibility(key_path):
-              find_specs = self.BuildFindSpecsFromRegistryArtifact(key_path)
-              artifact_group = self._find_specs_per_source_type[
-                  artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY]
-              artifact_group.extend(find_specs)
-
-        elif (source.type_indicator ==
-              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
-          # TODO: Handle Registry Values Once Supported in dfwinreg.
-          # https://github.com/log2timeline/dfwinreg/issues/98
-          logger.warning((
-              'Windows Registry values are not supported, extracting key: '
-              '"{0!s}"').format(source.key_value_pairs))
-
-          # TODO: move source.key_value_pairs iteration into
-          # BuildFindSpecsFromRegistryArtifact.
-
-          # Use set-comprehension to create a set of the source key paths.
-          source_key_paths = {
-              key_value['key'] for key_value in source.key_value_pairs}
-          for key_path in source_key_paths:
-            if self.CheckKeyCompatibility(key_path):
-              find_specs = self.BuildFindSpecsFromRegistryArtifact(key_path)
-              artifact_group = self._find_specs_per_source_type[
-                  artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY]
-              artifact_group.extend(find_specs)
-
-        elif (source.type_indicator ==
-              artifact_types.TYPE_INDICATOR_ARTIFACT_GROUP):
-          self._artifacts.remove(name)
-          for name_entry in set(source.names):
-            self._artifacts.append(name_entry)
-          self.BuildFindSpecs(environment_variables=environment_variables)
-
-        else:
-          logger.warning(
-              'Unsupported artifact definition source type: "{0:s}"'.format(
-                  source.type_indicator))
+      find_specs.extend(self._BuildFindSpecsFromArtifact(
```

Comment:
please split this into multiple lines, one function call per line to improve debugging.

Reply:
Done